### PR TITLE
fix: fix some of error handling

### PR DESCRIPTION
**Pull request by klu909**
## Context

adding some of missing error handling from queue and redis object which causes queue-serivce to exit upon error
## Objective

adding some of missing error handling from queue and redis object which causes queue-serivce to exit upon error
## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Comment by klu909** [comment]
still need to fix some tests

**Comment by klu909** [comment]
ideally we would also want to utilize retryStrategy and maxRetriesPerRequest from [ioredis](https://github.com/luin/ioredis#auto-reconnect) but seems like it was always using default value of 20 for maxRetriesPerRequest instead whatever value we provided

**Comment by DekusDenial** [comment]
looks fine to me, not sure if any breaking change from ioredis, cc @pritamstyz4ever

File: plugins/redis.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+redis.on('connecting', () => {
+    logger.info('Connecting to Redis.');
+});
+redis.on('connect', () => {
+    logger.info('Successfully connected to Redis');
+});
+redis.on('error', err => {
+    if (err.code === 'ECONNREFUSED') {
+        logger.error(`Could not connect to Redis: ${err.message}.`);
```
Comment (sagar1312) [inline plugins/redis.js:22]: What would be the implication of this in the queue service?
Can queue service operate without connecting to redis?

Comment (klu909) [reply plugins/redis.js:22]: ioredis will retry until `maxRetriesPerRequest`, we just catching this to log the error. Behavior will be same regardless

Comment (pritamstyz4ever) [reply plugins/redis.js:22]: How are we handling the cases of periodic job configurations getting lost?
Will they get fixed if the redis error is resolved?

**Review by pritamstyz4ever — COMMENTED** [review]
How are we handling the cases of periodic job configurations getting lost?
Will they get fixed if the redis error is resolved?

**Comment by klu909** [comment]
> How are we handling the cases of periodic job configurations getting lost? Will they get fixed if the redis error is resolved?

we have to manually add it back if it's lost

**Pull request merged**